DOC-501 (#577)
amplitude/amplitude-docs: content/collections/advanced-techniques/en/flag-prerequisites.md

updated_at: 1719252081
---
When running experiments or rolling out feature flags, some features may only be relevant if another feature is enabled. You can evaluate these dependencies first and use the results in your flag or experiment evaluation.

Amplitude Experiment lets you create dependencies based on prerequisite flags or experiments.

### Feature availability

This feature is available to Enterprise plan users who have purchased Amplitude Experiment. See the [pricing page](https://amplitude.com/pricing) for more details.

## Configuring flag prerequisites

You can configure flag prerequisites in the Dependencies card, located below the Overview card for experiments and below the Settings card for flags.

This card provides a summary of all flag dependencies, including prerequisite flags, experiments, mutual exclusions, and holdout groups, as well as a list of flags and experiments dependent on it.

1. To configure new prerequisites, select the edit icon.
2. In the Dependencies edit drawer, select _Add Dependency_ to add a new prerequisite flag or experiment.
3. Select the flag or experiment you want to create the dependency on.

	{{partial:admonition type='note'}}

> 	Flags and experiments are eligible to be used as a prerequisite if:

Reviewer: I think we can clean this up quite a bit as well. Active voice, present tense and all that.

> You can use a flag or experiment as a prerequisite if it meets the following criteria:
> 
> - The flag or experiment is in the same project
> - They use compatible evaluation modes. Flags and experiments that use local evaluation support prerequisites that use local evaluation. Flags and experiments that use remote evaluation support prerequisites of either evaluation type.

	- They're in the same project.
	- They've compatible evaluation modes. Local evaluation mode flags and experiments can only have local evaluation mode prerequisites, while remote evaluation mode flags and experiments can have both remote and local prerequisites.

> 	You can’t add a prerequisite that would create a circular dependency loop.

Reviewer: Need to dig deeper on this case. I assume it means you can't add things that are prerequisites of eachother.

	{{/partial:admonition}}

> 1. Then, select the variants that you want to be dependent on. Off is a special value that represents when users weren't included in the prerequisite flag or experiment.

Reviewer: ```suggestion
1. Select the variants that you want to be dependent on.`off` is a special value that represents when users weren't included in the prerequisite flag or experiment.
```

If you haven't, please look along at the UI as you go through these to ensure option names and casing match.

2. To finish adding the dependency, select Save.

## Workflow considerations

Before you activate a flag or start an experiment, ensure that prerequisite flags are active and variant assignment works as intended. You need to activate prerequisite flags and experiments before starting the experiment.

For flags and experiments with dependents, Amplitude prevents the following actions:

- Deleting a variant or changing the variant key of a variant that another flag or experiment depends on.

- Archiving that flag or experiment.

## Example with evaluation details

This example provides more information about how evaluation works when there are prerequisite flags.

> Example scenario: I want to ensure that my new feature flag (Flag-B) rolls out to users who have seen another feature (Flag-A). In Flag-B, I’ve added a dependency for the on variant of Flag-A and activated both flags.

Reviewer: I -> you

Frame this in the context of the person reading the article.

### Release groups

> Use flag prerequisites to build a primary feature with multiple sub-features. Sub-features require the primary feature to be on, unless a user is individually included as a tester in one of the sub-features. Targeting and bucketing applied to the primary feature are also applied to all sub-features that list the primary feature as a prerequisite.

Reviewer: ```suggestion
Use flag prerequisites to build a primary feature with multiple sub-features. Sub-features require the primary feature to be on, unless a you include an individual user as a tester in one of the sub-features. Targeting and bucketing options that apply to the primary feature apply to all sub-features that list the primary feature as a prerequisite.
```

active voice

Common use cases for release groups include:

- Actively developing large feature releases with many developers and teams.
- Provisioning users to primary SKUs with add-ons.
- Simplifying feature flag logic in the code.

![Diagram of example release group.](/docs/output/img/experiment/release-group.drawio.svg)

This example contains a primary-feature flag and sub-feature flags that list the primary feature as a prerequisite.

> The primary-feature flag targets all users where the user property premium is true with 100% allocation. Therefore, sub-features are only evaluated if the user has the required user property and meets the sub-feature's criteria—unless the user is individually included in the sub-feature's testers section.

Reviewer: code markup was useful here. we usually write property names and values this way.
